utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -70,4 +77,48 @@ class PerformanceMetricsAssessorImpl @Inject constructor(
       else -> HIGH_MEMORY_TIER
     }
   }
+
+  override fun computeCpuSnapshotAtCurrentTime(): CpuSnapshot {
+    return CpuSnapshot(
+      appTimeMillis = oppiaClock.getCurrentTimeMs(),
+      cpuTimeMillis = Process.getElapsedCpuTime(),
+      numberOfOnlineCores = getNumberOfOnlineCores()
+    )
+  }
+
+  override fun getRelativeCpuUsage(
+    firstCpuSnapshot: CpuSnapshot,
+    secondCpuSnapshot: CpuSnapshot
+  ): Double? {
+    if (
+      firstCpuSnapshot.isNewer(secondCpuSnapshot) ||
+      firstCpuSnapshot.doesNotHaveValidNumberOfOnlineCores() ||
+      secondCpuSnapshot.doesNotHaveValidNumberOfOnlineCores()
+    ) { return null }
+
+    val deltaCpuTimeMs = secondCpuSnapshot.cpuTimeMillis - firstCpuSnapshot.cpuTimeMillis
+    val deltaProcessTimeMs = secondCpuSnapshot.appTimeMillis - firstCpuSnapshot.appTimeMillis
+    val numberOfCores =
+      secondCpuSnapshot.numberOfOnlineCores + firstCpuSnapshot.numberOfOnlineCores / 2.0
```

Comment:
Parentheses are needed here for the computation to work correctly. Perhaps add a test to verify this exact case? :)

Reply:
Ah, yes. Actually one of the existing iterations -- `nonZeroDoubleValueTillTwoDecimalPoints` -- verifies this only. It was an oversight at my end to change the test result value from 0.5 to 0.4 when it failed. Fixed now.